bluesky/tests/test_est_time.py

```diff
@@ -0,0 +1,105 @@
+                                 {'action': 'unstage', 
+                                  'estimation':{'std_dev': float('nan'), 'time': float('nan')},
+                                  'object_name': 'motor1', 'time': {'start': 20, 'stop': 21.05}},
+                                 {'action': 'unstage', 
+                                  'estimation':{'std_dev': float('nan'), 'time': float('nan')},
+                                  'object_name': 'motor1', 'time': {'start': 22, 'stop': 22.95}}])
+                                                                #add 2 'stage/unstage' values for motor1
+
+    TelemetryUI.telemetry.extend([{'acquire_period': {'setpoint': 1}, 'acquire_time': {'setpoint': 1},
+                                  'action': 'trigger', 
+                                  'estimation': {'std_dev': float('nan'), 'time': 1.0},
+                                  'num_images': {'setpoint': 1}, 'object_name': 'det1',
+                                  'settle_time': {'setpoint': 0}, 'time': {'start': 20, 'stop': 21.05},
+                                  'trigger_mode': {'setpoint': 1}},
+                                 {'acquire_period': {'setpoint': 1}, 'acquire_time': {'setpoint': 1.0},
+                                  'action': 'trigger', 
+                                  'estimation': {'std_dev': float('nan'), 'time': 1.05},
+                                  'num_images': {'setpoint': 1}, 'object_name': 'det1',
+                                  'settle_time': {'setpoint': 0}, 'time': {'start': 22, 'stop': 22.95},
+                                  'trigger_mode': {'setpoint': 1}}]) #add 2 'trigger' values for det1
+
+    TelemetryUI.telemetry.extend([{'action': 'stage', 
+                                  'estimation':{'std_dev': float('nan'), 'time': float('nan')},
+                                  'object_name': 'det1', 'time': {'start': 20, 'stop': 21.05}},
+                                 {'action': 'stage', 
+                                  'estimation':{'std_dev': float('nan'), 'time': float('nan')},
+                                  'object_name': 'det1', 'time': {'start': 22, 'stop': 22.95}},
+                                 {'action': 'unstage', 
+                                  'estimation':{'std_dev': float('nan'), 'time': float('nan')},
+                                  'object_name': 'det1', 'time': {'start': 20, 'stop': 21.05}},
+                                 {'action': 'unstage', 
+                                  'estimation':{'std_dev': float('nan'), 'time': float('nan')},
+                                  'object_name': 'det1', 'time': {'start': 22, 'stop': 22.95}}])
+                                                                #add 2 'trigger' values for det1
+        
+
+def test_EpicsMotorEstTime():
+    #test the time estimation for epics motorlike devices.
+    setup_telemetry()
+    assert round(motor1.est_time.set(1,2,1,4).est_time,4) == 4.9975
```

Comment:
The standard way to do float comparisons in tests as `assert_almost_equal` or `assert_allclose`. You can find examples elsewhere in the bluesky tests.

Reply:
I will look at both of these options that I didn't find in my initial investigation of pytest.